### DISCUSSION TITLE
Using a custom renamed shell component

### DISCUSSION THREAD

**Discussion by pchemguy** [sqlpage/SQLPage#416] [Q&A] [answered]
I wanted to have login/logout buttons as menu-item elements, so I modified slightly the _shell_ component. Now, the modification appears to be well-localized, but to stay safe I renamed it to _shell_ex_ . The _shell_ component contains the "core" HTML structure and should only be used once. If I call my modified/renamed _shell_ex_ component, may it cause any issues with that page (it appears OK so far, and I am planning to post the details later)?

**Comment by lovasoa** [comment, marked as answer]
Hi !
  Indeed, and component [with a name that starts with `shell`](https://github.com/lovasoa/SQLpage/blob/main/src/render.rs#L342-L344) will be [treated as a shell component](https://github.com/lovasoa/SQLpage/blob/main/src/render.rs#L296-L325), and can be used in place of the default shell. SQLPage embeds two shells by default: [`shell`](https://github.com/lovasoa/SQLpage/blob/main/sqlpage/templates/shell.handlebars) and [`shell-empty`](https://github.com/lovasoa/SQLpage/blob/main/sqlpage/templates/shell-empty.handlebars).
 What modifications did you do to the shell ? If you think they could be useful to others, please open a pull request to merge them here !

**Reply by pchemguy** [reply]
Thanks for the clarification.

I am cleaning my mini demo that includes a number of features I explored. I will make a demo repo first, or perhaps, an folder inside examples. Then we can discuss how to proceed from there.
